narwhal/storage/src/certificate_store.rs

```diff
@@ -137,6 +149,12 @@ impl CertificateStore {
     /// Retrieves a certificate from the store. If not found
     /// then None is returned as result.
     pub fn read(&self, id: CertificateDigest) -> StoreResult<Option<Certificate>> {
+        fail::fail_point!("certificate-store-panic", |_| {
+            Err(RocksDBError(format!(
+                "Injected error in certificate store read"
+            )))
+        });
```

Comment:
I don't think injecting storage failures is a good idea atm. We previously discussed that if a storage failure happens in some code paths the only thing possible to do is crash the program. cc @lxfind @andll

Reply:
Was discussing this with Andrey and we are planning on testing this out in a test environment with an additional tool that restarts nodes that panic/crash to see how it recovers. Maybe we can replicate the data issue that is being investigated in [sui_testnet_ne_inc_94](https://mysten-labs.slack.com/archives/C04D8B4V63W/p1669837500152819) or discover some other edge cases?